metrics/src/metrics.rs

```diff
@@ -53,12 +53,39 @@ macro_rules! datapoint {
         point
         }
     };
+    ($name:expr, $($fields:tt)+) => {
+        $crate::submit($crate::datapoint!(@point $name, $($fields)+), log::Level::Debug);
```

Comment:
Can log_enabled! call be moved up to the macro level so that the right module path is used for filtering and so filtering occurs up front?

Reply:
the latest patch should take care of this. Thanks for spotting the issue.